src/cmd/input.go

```diff
@@ -11,14 +11,16 @@ import (
 var dataFile string
 
 func readInputConfig() {
+	viper.SetConfigType("yaml")
```

Comment:
IIRC `-f filename.yaml` wasn't working until I moved this up here from outside of the `case "-"` block.

Reply:
can't think of a reason for this to not be applied in certain conditions vs others, so it makes sense to move to the top.

Reply:
I think this is explicitly set to `yaml` below since `viper` has no way to infer the filetype from the stdin input.

[viper.SetConfigFile()](https://pkg.go.dev/github.com/spf13/viper@v1.16.0#SetConfigFile) defines the path, name and extension of the config file. Then `viper.ReadInConfig()` will know how to read it later.

If somebody is using a `.json` file I would assume viper would internally reset the configType below?

In any case, @rocktavious has mentioned that we should look into a better solution for reading user input in place of viper

Reply:
ya we need to fix this reading from input problems globally